src/mlpack/methods/reinforcement_learning/environment/acrobat.hpp

```diff
@@ -0,0 +1,324 @@
+    arma::colvec data;
+  };
+  /* 
+   * Implementation of action for cartpole
+   */
+  enum Action
+  {
+    negativeTorque,
+    zeroTorque,
+    positiveTorque,
+    // Track the size of the action space.
+    size
+  };
+
+   /**
+   * Construct a Cart Pole instance using the given constants.
+   *
+   * @param gravity gravity
+   * @param linkLength1 length of link 1.
+   * @param linkLength2 length of link 2.
+   * @param linkMass1 mass of link 1.
+   * @param linkMass2 mass of link 2.
+   * @param linkCom1 position of the center of mass of link 1.
+   * @param linkCom2 position of the center of mass of link 2.
+   * @param linkMoi moments of inertia for both link.
+   * @param maxVel1 max angular velocity of link1.
+   * @param maxVel2 max angular velocity of link2.
+   */
+  Acrobat(const double gravity = 9.81,
+          const double linkLength1 = 1.0,
+          const double linkLength2 = 1.0,
+          const double linkMass1 = 1.0,
+          const double linkMass2 = 1.0,
+          const double linkCom1 = 0.5,
+          const double linkCom2 = 0.5,
+          const double linkMoi = 1.0,
+          const double maxVel1 = 4*M_PI,
+          const double maxVel2 = 9*M_PI,
+          const double dt = 0.2) :
+      gravity(gravity),
+      linkLength1(linkLength1),
+      linkLength2(linkLength2),
+      linkMass1(linkMass1),
+      linkMass2(linkMass2),
+      linkCom1(linkCom1),
+      linkCom2(linkCom2),
+      linkMoi(linkMoi),
+      maxVel1(maxVel1),
+      maxVel2(maxVel2),
+      dt(dt)
+  { /* Nothing to do here */ }
+  /**
+    * Dynamics of the Acrobat System.
+    * To get reward and next state based on current
+    * state and current action .
+    * 
+    * @param state The current State
+    * @param action The action taken
+    * @param nextState The next state
+    * return reward -1
+    * Always return -1 reward
+    */
+  double Sample(const State& state,
+                 const Action& action,
+                 State& nextState)
+  {
+    Rk4(state, action, nextState);
+    nextState.Theta1() = Wrap(nextState.Theta1(), -M_PI, M_PI);
+
+    nextState.Theta2() = Wrap(nextState.Theta2(), -M_PI, M_PI);
+    nextState.AngularVelocity1() = Bound(nextState.AngularVelocity1(),
+                                             -maxVel1 , maxVel1);
+    nextState.AngularVelocity2() = Bound(nextState.AngularVelocity2(),
+                                             -maxVel2 , maxVel2);
+    return -1;
+  };
+  double Sample(const State& state, const Action& action)
+  {
+    State nextState;
+    return Sample(state, action, nextState);
+  }
+  State InitialSample() const
+  {
+    return State((arma::randu<arma::colvec>(4) - 0.5) / 5.0);
+  }
+  /**
+   *
+   * @param state The current State
+   */        
+  bool IsTerminal(const State& state) const
+  {
+    return bool (-cos(state.Theta1())-cos(state.Theta1()
+      + state.Theta2()) > 1.0);
+  }
```

Comment:
The goal of the game is to reach a height of one link as soon as possible. I have implemented this here. Now we cannot wait for long time for the convergence. I wanted a similar approach as of Shantao's to check convergence . 

    /**
     * Reaching running average return 35 is enough to show it works.
     * For the speed of the test case, I didn't set high criterion.
     */

https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/q_learning_test.cpp

Reply:
For the test this is fine, but we should also make sure we have something that can solve the env, e.g. is able to get a reward of ... over 50 consecutive trials. Also, can you also add a test in the form of https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/rl_components_test.cpp.

Reply:
@zoq my bad I missed the rl test previous time. However I have added this.
I am not clear about the 50 consecutive trails you are talking about.

Reply:
What I mean with 50 consecutive trails is that we should make sure the method we test against (QLearning) is able to solve the env with different starting conditions x times in a row.

Reply:
I have added it. I believe now it is ready to be merged :)

Reply:
@zoq Done :).

Reply:
@zoq Do you mind explaining me the reason to check for 50 consecutive trails. It takes huge amount of time to check for convergence. Ideally we do not have to give so much time for testing. Would you please explain me the where I am missing .

Reply:
You are right, we should not implement this as a unit test, but we should test this at least once, e.g. modify the test script you already have. I just like to make sure, the environment works as expected, and this is just another test I think might be helpful.

Reply:
@zoq it is ready to be merged.